upgrade/upgrade_windows.go

```diff
@@ -0,0 +1,25 @@
+// +build !cgo
+// +build upgrade
+
+package main
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+func buildAmalgamation(baseDir, buildFlags string) error {
+	args := []string{"/f", "Makefile.msc", "sqlite3.c"}
```

Comment:
Please check "gcc" can be used at the first.

Reply:
What command should be run if gcc is available? And how should I check if it is?
Sorry, I'm not developing on Windows, so I'm not sure of all the ways to build this, and what the typical toolchains would be.
The only thing I tried was installing Visual Studio and ActiveTcl, and then running the recommended `nmake` script. This worked for me. I didn't try MinGW or anything like that...

Reply:
Ehm, I was still waiting for some clarifications here...

Reply:
Still no feedback? :/

Reply:
Mostly like UNIX.

Reply:
Thanks for looking at it again!

So does running
```
$ sh configure
$ make sqlite3.c
```
work indeed?

In which case should we get rid of `upgrade_windows.go` altogether?
Or should both ways of building be available, either picked based on some CLI flag, or by trying one and checking for 'executable not found' or something to decide whether it should fall back to the other way?

Reply:
```
c:\msys64\usr\bin\bash.exe -c ./configure
make sqlite3.c
```

Reply:
Sounds good. I'll make windows and unix code symmetrical then, simply with default paths to `bash` and `make` different (and configurable). And will get rid of `nmake`, unless you think it's useful to leave it in there?

Reply:
(Is the `-c` necessary, by the way? Not that it makes much difference... just curious. I will now invoke configure this way indeed, and run `bash` by default.)

Reply:
Hey again! Any thoughts about the current implementation? Do you want me to do any more changes?